src/cryptonote_core/service_node_quorum_cop.cpp

```diff
@@ -216,14 +219,15 @@ namespace service_nodes
           if (!alive_for_min_time)
             break;
 
+          uint8_t obligations_height_hf_version = m_core.get_hard_fork_version(m_obligations_height);
```

Comment:
Putting this outside the loop doesn't seem right: the `hf_version` can change through the loop below.

Reply:
Yes, fixed